### ISSUE TITLE
Compute ranks when vectors are created 

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#1126]
We spend a significant amount of time updating ranks on freshly created vectors that could be cut down by inserting vectors with ranks.

We could also compute ranks with SQL without retrieving data altogether

**Comment by arteymix** [comment]
Once that is done, we can make vectors immutable like we did for analyses.

**Comment by ppavlidis** [comment]
Gemma 2 disposition: Won't fix
